gen-all-repos.py

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/python3
+# pylint: disable=invalid-name
+"""
+Generate all repo snapshot configuration files based on the definitions list
+"""
+
+import argparse
+import json
+import logging
+import subprocess
+import sys
+
+
+logging.basicConfig(level=logging.INFO, format='%(message)s')
+logger = logging.getLogger(__name__)
+
+
+def get_parser():
+    """
+    Create argument parser
+    """
+    parser = argparse.ArgumentParser(description=__doc__)
+    parser.add_argument(
+        '--definitions',
+        default='repo-definitions.json',
+        help='Path to the definitions list'
+    )
+    parser.add_argument(
+        '--output-dir',
+        default='./repo',
+        help='Path to the output directory'
+    )
+    parser.add_argument(
+        '-d', '--debug',
+        action='store_true',
+        default=False,
+        help='Enable debug logging'
+    )
+    return parser
+
+# pylint: disable=too-many-locals,too-many-branches
+def main():
+    """
+    Main function
+    """
+    parser = get_parser()
+    args = parser.parse_args()
+
+    if args.debug:
+        logger.setLevel(logging.DEBUG)
+
+    logger.info('Reading definitions from %s', args.definitions)
+    with open(args.definitions, encoding='utf-8') as file:
+        definitions = json.load(file)
+
+    for distro, repos in definitions.items():
+        for repo in repos:
+            cmd = ['./gen-repos.py', '--target-dir', args.output_dir]
```

Comment:
Please use `[sys.executable, 'gen-repos.py', '--target-dir', args.output_dir]`, this avoids leaving a virtual environment if applicable.